models/models.js

```diff
@@ -7,15 +7,41 @@ exports.selectTopics = () => {
   });
 };
 
-exports.selectArticles = (sort_by = "created_at") => {
-  queryStr = `SELECT articles.*, COUNT(articles.article_id)::INT AS comment_count FROM articles
-  JOIN users ON articles.author = users.username
+exports.selectArticles = (
+  sort_by = "created_at",
+  userOrder = "desc",
+  topic
+) => {
+  console.log(sort_by);
+  const validColumns = [
+    "title",
+    "topic",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
 
-  LEFT JOIN comments ON comments.article_id = articles.article_id
-  GROUP BY articles.article_id
-  ORDER BY articles.created_at DESC;`;
+  if (!validColumns.includes(sort_by)) {
+    return Promise.reject({ status: 400, msg: "invalid sort query" });
+  }
+  const validOrder = ["desc", "asc"];
+  if (!validOrder.includes(userOrder)) {
+    return Promise.reject({ status: 400, msg: "invalid sort query" });
+  }
+  let queryStr = `SELECT articles.*, COUNT(articles.article_id)::INT AS comment_count FROM articles
+  JOIN users ON articles.author = users.username LEFT JOIN comments ON comments.article_id = articles.article_id
+  `;
```

Comment:
I know the instructions might be a little confusing, but given the data you are actually responding with here, do you need to be JOINing the `users` table here? (It seems like you aren't using any info from that table for anything useful, so maybe not...)